packages/backend-common/src/database/DatabaseManager.ts

```diff
@@ -63,40 +54,12 @@ export type LegacyRootDatabaseService = {
 };
 
 /**
- * Manages database connections for Backstage backend plugins.
- *
- * @public
- * @remarks
- *
- * The database manager allows the user to set connection and client settings on
- * a per pluginId basis by defining a database config block under
- * `plugin.<pluginId>` in addition to top level defaults. Optionally, a user may
- * set `prefix` which is used to prefix generated database names if config is
- * not provided.
+ * Testable implementation class for {@link DatabaseManager} below.
```

Comment:
If it's only about construction you could also have a `forTesting` static factory or similar on the main implementation, and mark it as `@internal`

Reply:
Oh good point - can i do that after though? This isn't visible to the outside either way